code/mob/living/silicon/robot.dm

```diff
@@ -996,22 +996,21 @@
 			src.compborg_lose_limb(PART)
 
 	emag_act(var/mob/user, var/obj/item/card/emag/E)
-		if(isshell(src) || src.part_head.ai_interface)
-			boutput(user, SPAN_ALERT("Emagging an AI shell wouldn't work, [his_or_her(src)] laws can't be overwritten!"))
-			return 0 //emags don't do anything to AI shells
 		if (!src.emaggable)
 			boutput(user, SPAN_ALERT("You try to swipe your emag along [src]'s interface, but it grows hot in your hand and you almost drop it!"))
-			return FALSE
-
+			return 0
 		if (!src.emagged)	// trying to unlock with an emag card
 			if (src.opened && user) boutput(user, "You must close the cover to swipe an ID card.")
 			else if (src.wiresexposed && user) boutput(user, SPAN_ALERT("You need to get the wires out of the way."))
 			else
 				if (user)
-					boutput(user, "You emag [src]'s interface.")
-				src.visible_message(SPAN_ALERT("<b>[src]</b> buzzes oddly!"))
-				logTheThing(LOG_STATION, src, "[key_name(src)] is emagged by [key_name(user)] and loses connection to rack. Formerly [constructName(src.law_rack_connection)]")
-				src.mind?.add_antagonist(ROLE_EMAGGED_ROBOT, respect_mutual_exclusives = FALSE, source = ANTAGONIST_SOURCE_CONVERTED)
+					boutput(user, "You short out [src]'s interface lock.")
+				src.emagged = TRUE
+				if(!src.brainexposed)
+					src.brainexposed = TRUE
+					src.visible_message(SPAN_ALERT("<b>[src]</b>' head compartment pops open!"))
+				boutput(src, SPAN_ALERT("Your interface lock has been emagged!"))
+				elecflash(src)
 				update_appearance()
 				return 1
 			return 0
```

Comment:
should add some feedback to person using the emag if they're already emagged imho. Maybe always open them on emag use?